### PR TITLE
Validate blending on color attachment without blending support VU 02023

### PR DESCRIPTION
**Pull request by djdeath**
We got this bug report from the Angle team at Google : https://gitlab.freedesktop.org/mesa/mesa/-/issues/2743

After checking the spec I realized this should have been reported as an error in the validation layer but we're missing this VU.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -980,6 +980,25 @@ bool CoreChecks::ValidatePipelineUnlocked(const PIPELINE_STATE *pPipeline, uint3
...
+            const auto attachment = subpass_desc->pColorAttachments[i].attachment;
+            if (attachment != VK_ATTACHMENT_UNUSED) {
+                VkFormat format = pPipeline->rp_state->createInfo.pAttachments[attachment].format;
+                VkFormatProperties properties;
+                DispatchGetPhysicalDeviceFormatProperties(physical_device, format, &properties);
+
+                if (pPipeline->graphicsPipelineCI.pRasterizationState &&
+                    !pPipeline->graphicsPipelineCI.pRasterizationState->rasterizerDiscardEnable &&
+                    pPipeline->attachments[i].blendEnable &&
+                    !(properties.optimalTilingFeatures & VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BLEND_BIT)) {
```
Comment (Tony-LunarG) [inline layers/core_validation.cpp:992]: Do you know for sure that tiling is optimal here?

Comment (djdeath) [reply layers/core_validation.cpp]: You're right, I can only validate this properly at pipeline bind time.

**Comment by djdeath** [comment]
This new version will validate this VU at bind pipeline time and use the appropriate format features bits.

**Comment by Tony-LunarG** [comment]
Passed CI, and LGTM.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -4617,6 +4617,57 @@ static const char *GetPipelineTypeName(VkPipelineBindPoint pipelineBindPoint) {
...
+
+    if (fb_state) {
+        auto subpass_desc = &pipeline_state->rp_state->createInfo.pSubpasses[pipeline_state->graphicsPipelineCI.subpass];
+
+        for (size_t i = 0; i < pipeline_state->attachments.size(); i++) {
+            const auto attachment = subpass_desc->pColorAttachments[i].attachment;
+            if (attachment == VK_ATTACHMENT_UNUSED) continue;
+
+            const IMAGE_VIEW_STATE *imageview_state = GetImageViewState(fb_state->createInfo.pAttachments[attachment]);
+            if (!imageview_state) continue;
```
Comment (Tony-LunarG) [inline layers/core_validation.cpp:4632]: Actually, this code doesn't take into account imageless framebuffers, where the createInfo.pAttachments is null and image views are supplied at BeginRenderPass.  The CTS test dEQP-VK.imageless_framebuffer.color will crash here if run with validation.

Comment (djdeath) [reply layers/core_validation.cpp:4632]: Oh thanks for pointing this out.

Let me test this change a bit and open an PR : https://github.com/djdeath/Vulkan-ValidationLayers/commit/096ec16f6f839c6378497f173ae8a243459b7be7

**Pull request merged**